sips/cluster_consensus.md

```diff
@@ -0,0 +1,127 @@
+
+```mermaid
+flowchart RL
+	subgraph MessageID
+		Domain
+		ValidatorPublicKey
+		Role
+	end
+```
+
+to
+
+
+```mermaid
+flowchart RL
+	subgraph MessageID
+		Domain
+		OperatorCluster
+		Role
+	end
+```
+
+### QBFT Controller
+
+The current `QBFTController` structure allows only one consensus instance at a time. This must change, extending the `QBFTController` into a router of messages for the different instances.
+
+### DutyRunner & QBFT Controller
+
+Since the `DutyRunner` will not have its specific `QBFTController`, it must have a way to start a consensus instance and receive its decided value. For that, we propose applying the observer design pattern by which the `DutyRunner` (observer) can be updated upon a `QBFT Instance` (observable) termination.
+
+We suggest that the `DutyRunner` holds a reference to the operator's `QBFTController` to start a `QBFT Instance` and observe it.
+
+```go
+func (r *DutyRunner) execute() {
+	r.QBFTController.StartConsensus(r.committee, r.duty, r)
+}
+
+func (c *QBFTController) StartConsensus(committee []types.Operator, duty types.Duty, observer Observer) {
+	instance := c.StartInstance(committee, duty)
+	instance.registerObserver(observer)
+}
+```
+
+## Drawbacks
+
+- Several duties will depend on the same consensus execution. Thus, its failure will imply many attestation misses. Nonetheless, this will also make the operator more scalable due to the overall reduction in exchanged messages and processing costs.
+
+## Extra improvements
+
+- If multiple partial signatures contained in a merged message refer to the same root (i.e. validators in the same Ethereum committee), the signatures can be verified using batch verification.
+
+## Message Validation
+
+This duties transformation requires similar changes in message validation, namely:
+- Different consensus executions are tagged by the `MessageID`. This change would be propagated with no further issues. However, the `MessageID` is used to get the validator's public key and the duty's role which are used as an ID to store the consensus state. This must be changed to use the operators' committee and the duty's role, or even simply the `MessageID`.
+- Message validation limits the number of attestation duties per validator by using the validator's public key contained in the `MessageID`. This is no longer possible. A new limitation can be accomplished by checking the number of validators a cluster of operators is assigned to. If this number is less than 32 (the number of slots in an epoch), then we can limit the number of attestation duties of such cluster per epoch. The only exception would be if such a cluster is assigned to a sync committee duty (considering that we will indeed merge attestations and sync committee duties altogether in the same consensus execution).
```

Comment:
you also update message validation in knowledge base
should be on a seperate branch...
Also google sheet should be versioned

Reply:
Sure. Will do it after all the changes here are agreed on.